modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java

```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
         String successMsg = String.format(
-                "Managed service %s unregistered success.", managedServiceName);
+                "Update registered service %s with Url %s", id, oclLocation);
+        log.info(successMsg);
         return Response.successResponse(successMsg);
     }
 
+    /**
+     * Unregister registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs unregister registered service.")
+    @Operation(description = "Unregister registered service using id.")
+    @DeleteMapping("/register/service/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional
+    public Response unregister(@PathVariable("id") String id) {
+        log.info("Unregister registered service using id {}", id);
+        registerService.unregisterService(id);
+        String successMsg = String.format(
+                "Unregister registered service using id %s success.", id);
+        log.info(successMsg);
+        return Response.successResponse(successMsg);
+    }
+
+
+    /**
+     * List registered service with query params.
+     *
+     * @param cspName     name of cloud service provider.
+     * @param serviceName name of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to list the managed services.")
+    @Operation(description = "List registered service with query params.")
+    @GetMapping(value = "/register/services",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response listRegisteredService(
+            @RequestParam(name = "cspName", required = false) String cspName,
+            @RequestParam(name = "serviceName", required = false) String serviceName) {
+        RegisterServiceQuery query = new RegisterServiceQuery();
+        query.setCspName(cspName);
+        query.setServiceName(serviceName);
+        log.info("List registered service with query model {}", query);
+        List<RegisterServiceEntity> serviceEntities = registerService.listRegisteredService(query);
+        String successMsg = String.format("List registered service with query model %s "
+                + "success.", query);
+        log.info(successMsg);
+        return Response.successResponse(serviceEntities);
+    }
+
+
+    /**
+     * Get registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to get detail the registered services.")
+    @Operation(description = "Get registered service using id.")
+    @GetMapping(value = "/register/service/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response detail(@PathVariable("id") String id) {
```

Comment:
we must return OCL object of the service.

Reply:
Ok，I will set the OCL object into the Response directly.